opendj-server-legacy/src/main/java/org/opends/guitools/controlpanel/ui/SimplifiedViewEntryPanel.java

```diff
@@ -1640,19 +1641,14 @@ private void updatePanel(ObjectClassValue newValue)
         }
         if (isPassword(attrName))
         {
-          List<String> newPwds = getNewPasswords(attrName);
-          if (newPwds.equals(lastUserPasswords.get(attrName)))
+          // Do NOT rebuild password attribute from the editor here, carry over the original values so we do not corrupt them.
+          Attribute oldAttr = searchResult.getAttribute(attrName);
+          if (oldAttr != null && !oldAttr.isEmpty())
```

Comment:
Why not using the original logic? Was there a specific reason for changing these lines?